### ISSUE TITLE
Toolbar buttons flicker on view change

### ISSUE THREAD

**Issue by ghost** [sequelpro/sequelpro#789]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=789
Reporter:gene...@gmail.com Date:2010-08-14 05:17:45 Status:Accepted

<b>What steps will reproduce the problem?</b>
1. Connect to server and choose a database.
2. Choose any table in the left sidebar (preferably a table with a few rows).
3. Using toolbar buttons, try to switch between Structre/Content/Relations/Table Info/Query dialogs. Buttons will flicker.

<b>What is the expected output? What do you see instead?</b>
Buttons shouldn't flicker

<b>What version of Sequel Pro are you using? What version of MySQL are you</b>
<b>connecting to on the server?</b>
0.9.8.1, MySQL 5.1.49, Mac OSX 10.5.8

<b>Please provide any additional information below.</b>
It seems that toolbar buttons are disabled when loading table data. So, when user switches between those views, they are flickering because of temporarily disabled state. Nevertheless, there should be a better way to handle this.

My suggestion:
1) At first, don't grey them out, just disable buttons functionality so 
2) It loading time seems to last more than 0.3 sec, grey them out also (as in current behavior).

0.3 sec in this example could be anything, it just set some threshold so greying out will not be misinterpreted as flickering.

Other option: don't disable button functionality at all, let user click them and do what he wants. In this case, switching between views should terminate current rendering process and jump in a new one instead.

**Comment by fulldecent** [comment]
I think current behavior is correct. You cannot click the button while switching tables. Recommending close as WONTFIX.

**Comment by abhibeckert** [comment]
I disagree, it shouldn't flicker and also it should be possible to click those buttons while loading the next table.
